lib/pos_consensus_controller.go

```diff
@@ -211,3 +211,63 @@ func (cc *ConsensusController) HandleHeader(pp *Peer, msg *MsgDeSoHeader) {
 func (cc *ConsensusController) HandleBlock(pp *Peer, msg *MsgDeSoBlock) {
 	// TODO
 }
+
+func (cc *ConsensusController) fetchValidatorListsForBlocks(blocks []*MsgDeSoHeader) ([]consensus.BlockWithValidatorList, error) {
+	// Create a map to cache the validator set entries by epoch number. Two blocks in the same epoch will have
+	// the same validator set, so we can use an in-memory cache to optimize the validator set lookup for them.
+	validatorSetEntriesByEpochNumber := make(map[uint64][]*ValidatorEntry)
+
+	// Fetch the validator set at each block
+	blocksWithValidatorLists := make([]consensus.BlockWithValidatorList, len(blocks))
+	for ii, block := range blocks {
+		blockHash, err := block.Hash()
+		if err != nil {
+			return nil, errors.Errorf("error computing block hash: %v", err)
+		}
+
+		utxoViewAtBlock, err := cc.blockchain.getUtxoViewAtBlockHash(*blockHash)
+		if err != nil {
+			return nil, errors.Errorf("error fetching UtxoView for block: %v", err)
+		}
```

Comment:
So this function gets us half way there, but I think what we want to do is create a single utxo view struct without connecting any uncommitted blocks. We should have all the snapshot data available in the database that we want and connecting parents only really helps us figure out "current epoch". 

What I think we want to do is the following:
1. create a single vanilla utxo view `NewUtxoView(cc.blockchain.db, cc.blockchain.params, nil, cc.blockchain.snapshot, nil)` or maybe not nil for event manager, but that's up to you.
2. `currentEpochEntry, err := utxoView.GetCurrentEpochEntry()` 
3. For each block,
-  if `currentEpochEntry.ContainsBlockHeight(block.Height)`, use current epoch to compute snapshotAtEpochNumber
-  if not , we know must be in the next epoch. We can simulate the epoch using a function I wrote, basically just ripped out of the rollover logic,  called `computeNextEpochEntry`. Confirm block is in the next epoch and lookup snapshotAtEpochNumber for that computed epoch.
- Look up all validators using `GetAllSnapshotValidatorSetEntriesByStakeAtEpochNumber` (all in my pos-process-block stack)